mungegithub/mungers/milestone-maintainer.go

```diff
@@ -201,219 +377,271 @@ func ignoreObject(obj *github.MungeObject, activeMilestone string) bool {
 	return milestone != activeMilestone
 }
 
-// notificationState returns the state required to ensure the munger's
-// notification is kept current. If a nil return value is returned,
-// no action should be taken.
-func notificationState(obj *github.MungeObject, comment *c.Comment, botName string, gracePeriod, warningInterval time.Duration) (string, *milestoneNotification) {
-	notification := c.ParseNotification(comment)
-
-	kindLabel, priorityLabel, sigLabels, labelErrors := checkLabels(obj.Issue.Labels)
+type issueChangeConfig struct {
+	state             milestoneState
+	enabledSections   sets.String
+	templateArguments map[string]interface{}
+}
 
-	labelsComplete := len(labelErrors) == 0
-	if labelsComplete {
-		return milestoneLabelsCompleteLabel, labelsCompleteState(obj.Issue, notification, kindLabel, priorityLabel, sigLabels)
+// newIssueChange computes the changes required to modify the state of
+// the issue to reflect the milestone process. If a nil return value
+// is returned, no action should be taken.
+func newIssueChange(obj *github.MungeObject, mungerConfig *MilestoneMaintainer) *issueChange {
+	icc := newIssueChangeConfig(obj, mungerConfig)
```

Comment:
Why is it an issue to expose the `issueChangeConfig` struct?   As per a previous exchange , `newIssueChangeConfig` allows early exit which makes it hard to return the computed message body directly.  While I'm sure I could devise a way to return the computed message body (using `defer`?), that doesn't seem as clean to me.  Is there a strategy you would prefer?

Reply:
I was just thinking that the function could return `icc.state, icc.messageBody()` instead of `icc`, or something like that. Its not an issue to expose the struct outside the function, but if it isn't needed outside the function it seems cleaner to me to encapsulate it in the function. 
It doesn't really matter though, this is just a nit so feel free to disregard.